File: openmetadata-ui/src/main/resources/ui/src/antlr/SplitListener.js
/*
 *  Copyright 2022 Collate.
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *  http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
import FqnListener from '../generated/antlr/FqnListener';

export default class SplitListener extends FqnListener {
  constructor() {
    super();
    this.xs = [];
  }

  // Enter a parse tree produced by FqnParser#unquotedName.
  enterUnquotedName(ctx) {
    this.xs.push(ctx.getText());
  }

  // Enter a parse tree produced by FqnParser#quotedName.
  enterQuotedName(ctx) {
    this.xs.push(ctx.getText());
  }

  split() {
    return this.xs;
  }
}

File: ingestion/src/metadata/antlr/split_listener.py
#  Copyright 2025 Collate
#  Licensed under the Collate Community License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#  https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/LICENSE
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
"""
Helper class to handle FQN splitting logic
"""
from metadata.generated.antlr.EntityLinkListener import EntityLinkListener
from metadata.generated.antlr.EntityLinkParser import EntityLinkParser
from metadata.generated.antlr.FqnListener import FqnListener
from metadata.generated.antlr.FqnParser import FqnParser


class FqnSplitListener(FqnListener):
    def __init__(self):
        self._list = []

    def enterQuotedName(self, ctx: FqnParser.QuotedNameContext):
        self._list.append(ctx.getText())

    def enterUnquotedName(self, ctx: FqnParser.UnquotedNameContext):
        self._list.append(ctx.getText())

    def split(self):
        return self._list


class EntityLinkSplitListener(EntityLinkListener):
    def __init__(self):
        self._list = []

    def enterNameOrFQN(self, ctx: EntityLinkParser.NameOrFQNContext):
        self._list.append(ctx.getText())

    def enterEntityType(self, ctx: EntityLinkParser.EntityTypeContext):
        self._list.append(ctx.getText())

    def enterEntityField(self, ctx: EntityLinkParser.EntityFieldContext):
        self._list.append(ctx.getText())

    def split(self):
        return self._list
